LayoutTests/imported/blink/fast/forms/search-popup-crasher.html

```diff
@@ -17,7 +17,7 @@
 
 <p>This page tests that a page with a search popup does not crash, as reported in <a href='https://bugs.webkit.org/show_bug.cgi?id=37141'>this bug</a>. Below is a search input that the test automatically fills then submits. Pass if this does not crash.</p><br><br>
 <FORM action="" id="searchForm">
-<INPUT autosave="my.search" id="searchBox" type="search" results="10" value=""><br>
+<INPUT name="my.search" id="searchBox" type="search" results="10" value=""><br>
```

Comment:
I'm now worried that we may basically be introducing a feature here: auto saving for the first time. Apparently it has been broken for a long time. And even when we had it, websites had to opt in and almost none did. Now we will be saving on any form that uses the `name` attribute. I guess other browsers are doing that but does the UI in Safari for clearing his true work? Does this work automatically in other WebKit contexts? Do they offer a way to clear history? Does this create fingerprinting risk?

I‘m not an expert on all of this but I think it might require a little discussion with other folks at Apple.

Reply:
These are good questions. Actually I have researched a bit (since I am new to this code) on these when I made this patch, and here are the answers:
> does the UI in Safari for clearing this true work?  

We have SPI to fetch/remove this type of data; internal clients like Safari should be able to fetch/remove it.

> Does this work automatically in other WebKit contexts? Do they offer a way to clear history?

We don't have API for public clients to fetch/clear it. As we should start to save this data to disk, I think we should provide a way for WebKit client to clear it. I will make a followup to promote the SPI to API.

> Does this create fingerprinting risk?

This should not create fingerprinting risk because it is not available for web client (cannot be queried with Web API).
But we do have a problem that we don't separate this data by WebsiteDataStore (user might see the same search items when they load page in different sessions), and I made a patch at https://github.com/WebKit/WebKit/pull/9948 to fix this.

Reply:
It may not be enough to deal with just the WebKit project part of this, making sure that clients _could_ offer a way to clear this. I think we might to design WebKit API so that clients let us know they have a way to clear it and only then do we save it. Having hidden data that is a history of what you did on a site is a potential privacy problem if that old data shows up later.

Reply:
One way to do this would be to offer a 'savesSearchElementHistory' or 'shouldSaveSearchElementHistory' property on WKPreferences, which defaulted to false.

A webpage can already trigger a similar effect with [a few lines of LocalStorage code](https://stackoverflow.com/questions/63546745/save-input-search-history). But maybe triggering the effect implicitly and in a new circumstance warrants opt-in logic.

Reply:
> I think we might to design WebKit API so that clients let us know they have a way to clear it and only then do we save it.

Generally, when a data type (normally comes with a new feature) is introduced, the WebKit clients might not realize in time that they need to turn on a new flag, or they might not check whether the data type can be cleared before turning on the flag (likely not all clients care about privacy as we do).
So I tend to think it should be WebKit's responsibility to ensure there is a way to clear the data before introducing a new type, as Geoff mentioned in https://github.com/WebKit/WebKit/pull/10104#pullrequestreview-1298367093.

> One way to do this would be to offer a 'savesSearchElementHistory' or 'shouldSaveSearchElementHistory' property on WKPreferences, which defaulted to false.

For search field history specifically, I think there are two cases:
1. For WebKit clients who have history saved on disk (as I am not sure when the feature is broken and assume it works when first introduced), they are already in the situation of hidden data (cause no API to clear it), so it may not help if we turn off the feature in new builds.
2. For WebKit clients who have no history saved on disk due to "directory is not created" issue, they will start to have data saved on disk, and I have promoted the SPI to API in https://github.com/WebKit/WebKit/pull/10104, so the client should be able to clear it in new builds.

So I think `shouldSaveSearchElementHistory` might not help much for the privacy issue; it may be more useful if we know some WebKit clients just don't want the feature at all.

> A webpage can already trigger a similar effect with [a few lines of LocalStorage code](https://stackoverflow.com/questions/63546745/save-input-search-history). But maybe triggering the effect implicitly and in a new circumstance warrants opt-in logic.

It's true similar effect could be trigger with `LocalStorage` or other type of storage (and I think google does this). But I think the search field history is more of a user-facing feature as the data is shared across sites. For example, if you type 'iPhone 14' in apple.com, 'iPhone 14' may show up in history of search field on iCloud.com (not visible to iCloud.com).